### DISCUSSION TITLE
How Can Create Ant design Persian Calender (jalali)

### DISCUSSION THREAD

**Discussion by mahdiporkar** [ant-design/ant-design#51312] [General]
### What problem does this feature solve?

i saw in document we can set local fa but this cause title of calender change to persian but we need calender change to jalali 

### What does the proposed API look like?

.

**Comment by petercat-assistant** [comment]
To create a Persian (Jalali) calendar using Ant Design, you need to customize the calendar component to support the Jalali calendar system. Ant Design's default calendar component supports localization, but it might not fully support the Jalali calendar out of the box.

Here's a general approach to achieve this:

1. **Localization**: Ensure that you have set the locale to Persian (`fa`) in your Ant Design configuration. This will change the language of the calendar to Persian.

2. **Jalali Calendar Support**: You might need to integrate a library that supports the Jalali calendar system. One popular library for handling Jalali dates in JavaScript is `moment-jalaali` or `jalaali-js`. You can use this library to convert Gregorian dates to Jalali dates and vice versa.

3. **Custom Calendar Component**: You may need to create a custom calendar component that uses the Jalali date library to render dates correctly. This involves overriding the date rendering logic to display Jalali dates.

4. **Date Conversion**: Ensure that any date inputs and outputs are correctly converted between Gregorian and Jalali as needed.

Here's a basic example of how you might start integrating Jalali support:

```javascript
import React from 'react';
import { Calendar } from 'antd';
import moment from 'moment';
import 'moment/locale/fa';
import momentJalaali from 'moment-jalaali';

// Set locale to Persian
moment.locale('fa');
momentJalaali.loadPersian({ dialect: 'persian-modern' });

const JalaliCalendar = () => {
  const dateCellRender = (value) => {
    // Convert Gregorian date to Jalali
    const jalaliDate = momentJalaali(value).format('jYYYY/jM/jD');
    return <div>{jalaliDate}</div>;
  };

  return (
    <Calendar dateCellRender={dateCellRender} />
  );
};

export default JalaliCalendar;
```

This example demonstrates how to render Jalali dates in the calendar. You might need to further customize the component to fully meet your requirements.

If you are looking for a built-in feature in Ant Design to support Jalali calendars, it might not be available yet. You can consider contributing to the Ant Design project or requesting this feature if it's important for your use case.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by mahdiporkar** [comment]
can you tell me for date picker how can implement it ? 
thanks

**Comment by petercat-assistant** [comment]
To create a Persian (Jalali) calendar using Ant Design, you will need to customize the calendar component to support the Jalali calendar system. Ant Design's default calendar component supports localization, but it may not fully support the Jalali calendar out of the box. Here's a general approach to achieve this:

1. **Localization**: Ant Design provides localization support, which you can use to change the language to Persian. However, this only changes the text and not the calendar system itself.

2. **Jalali Calendar Integration**: You will need to integrate a Jalali calendar library with Ant Design's calendar component. One popular library for Jalali date manipulation is `moment-jalaali`. You can use this library to handle Jalali dates.

3. **Custom Calendar Component**: You might need to create a custom calendar component that uses `moment-jalaali` to display dates in the Jalali format. This involves overriding the date rendering logic of the Ant Design calendar component.

Here's a basic example of how you might start integrating `moment-jalaali`:

```javascript
import React from 'react';
import { Calendar } from 'antd';
import moment from 'moment-jalaali';

// Initialize moment-jalaali
moment.loadPersian({ dialect: 'persian-modern' });

const JalaliCalendar = () => {
  const dateCellRender = (value) => {
    // Convert the date to Jalali
    const jalaliDate = moment(value).format('jYYYY/jM/jD');
    return <div>{jalaliDate}</div>;
  };

  return (
    <Calendar dateCellRender={dateCellRender} />
  );
};

export default JalaliCalendar;
```

This example demonstrates how to render Jalali dates in the calendar cells. You may need to further customize the component to fully support all features of the Jalali calendar.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
